core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -203,32 +203,46 @@ ManifestReader<F> scanMetrics(ScanMetrics newScanMetrics) {
   }
 
   CloseableIterable<ManifestEntry<F>> entries() {
-    if ((rowFilter != null && rowFilter != Expressions.alwaysTrue())
-        || (partFilter != null && partFilter != Expressions.alwaysTrue())
-        || (partitionSet != null)) {
+    return entries(false /* all entries */);
+  }
+
+  private CloseableIterable<ManifestEntry<F>> entries(boolean onlyLive) {
+    if (hasRowFilter() || hasPartitionFilter() || partitionSet != null) {
```

Comment:
Do we need to consider the condition
```
!partitionSet.isEmpty()
```

Reply:
I believe an empty `partitionSet` indicates there is no match and acts similar to `alwaysFalse`. We could add a branch for cases when the filters are always false or the matching partition set is empty, but we would have to analyze the usages first to make sure those are possible scenarios.

Reply:
Thanks.